sso-kit-starter/src/main/java/com/vaadin/sso/starter/OidcLogoutTokenValidator.java

```diff
@@ -0,0 +1,220 @@
+            invalidClaims.put(LogoutTokenClaimNames.SID, sid);
+        }
+
+        // 6. Verify that the Logout Token contains an EVENTS Claim whose value
+        // is JSON object containing the member name
+        // http://schemas.openid.net/event/backchannel-logout.
+        final var events = token.getClaimAsMap(LogoutTokenClaimNames.EVENTS);
+        if (events == null || !events.containsKey(BC_LOGOUT_EVENT)) {
+            invalidClaims.put(LogoutTokenClaimNames.EVENTS, events);
+        }
+
+        // 7. Verify that the Logout Token does not contain a nonce Claim.
+        final var nonce = token.getClaim(IdTokenClaimNames.NONCE);
+        if (nonce != null) {
+            invalidClaims.put(IdTokenClaimNames.NONCE, nonce);
+        }
+
+        // 8. Optionally verify that another Logout Token with the same JTI
+        // value has not been recently received.
+        // TODO
+
+        // 9. Optionally verify that the IDD Logout Token Claim matches the ISS
+        // Claim in an ID Token issued for the current session or a recent
+        // session of this RP with the OP.
+        // TODO
+
+        // 10. Optionally verify that any SUB Logout Token Claim matches the SUB
+        // Claim in an ID Token issued for the current session or a recent
+        // session of this RP with the OP.
+        // TODO
+
+        // 11. Optionally verify that any SID Logout Token Claim matches the SID
+        // Claim in an ID Token issued for the current session or a recent
+        // session of this RP with the OP.
+        // TODO
+
+        if (!invalidClaims.isEmpty()) {
+            LOGGER.warn("Logout token validation failed because of invalid "
+                    + "claims: {}", invalidClaims);
+            return OAuth2TokenValidatorResult
+                    .failure(invalidLogoutTokenClaims(invalidClaims));
+        } else {
+            LOGGER.debug("Logout token validation succeded");
+            return OAuth2TokenValidatorResult.success();
+        }
+    }
+
+    /**
+     * Sets the maximum acceptable clock skew. The default is 60 seconds. The
+     * clock skew is used when validating the {@link JwtClaimNames#EXP exp} and
+     * {@link JwtClaimNames#IAT iat} claims.
+     *
+     * @param clockSkew
+     *            the maximum acceptable clock skew
+     */
+    public void setClockSkew(Duration clockSkew) {
+        Assert.notNull(clockSkew, "clockSkew cannot be null");
+        Assert.isTrue(clockSkew.getSeconds() >= 0, "clockSkew must be >= 0");
+        this.clockSkew = clockSkew;
+    }
+
+    /**
+     * Sets the {@link Clock} used in {@link Instant#now(Clock)} when validating
+     * the {@link JwtClaimNames#EXP exp} and {@link JwtClaimNames#IAT iat}
+     * claims.
+     *
+     * @param clock
+     *            the clock
+     */
+    public void setClock(Clock clock) {
+        Assert.notNull(clock, "clock cannot be null");
+        this.clock = clock;
+    }
```

Comment:
I see that theses setters are not in use right now. Who are supposed to use them

Reply:
True, right now we're hardcoding the validator instance into the filter. Still, once we do it properly (i.e. with a validator factory) they will be useful to set e.g. a non-UTC clock (which is the default). So I would leave those there, ready for that 🚀

Reply:
Got it!